Client/Frontend/Browser/BrowserViewController/BrowserViewController+WKNavigationDelegate.swift

```diff
@@ -319,6 +319,17 @@ extension BrowserViewController: WKNavigationDelegate {
 
     if navigationAction.targetFrame?.isMainFrame == true,
       BraveSearchManager.isValidURL(url) {
+
+      // Add Brave Search headers if Rewards is enabled
+      if !isPrivateBrowsing && rewards.isEnabled && navigationAction.request.allHTTPHeaderFields?["X-Brave-Ads-Enabled"] == nil {
+        decisionHandler(.cancel, preferences)
+
+        var modifiedRequest = URLRequest(url: url)
+        modifiedRequest.setValue("1", forHTTPHeaderField: "X-Brave-Ads-Enabled")
+        tab?.loadRequest(modifiedRequest)
+        return
```

Comment:
Isn't this gonna prevent the logic below related to Google fallback? In particular, aren't we relying on `tab?.injectResults()` being called in all cases to un-hide the results?

Reply:
It should get there at the second loop. First loop adds the header, then second loop handles the fallback.
On iOS we have to cancel and reload request in order to add headers, we can test it when new builds are uploaded